content/shmem_team_broadcast.tex

```diff
@@ -0,0 +1,90 @@
+\apiargument{IN}{source}{A symmetric data object that can be of any data type
+    that is permissible for the \dest{} argument.}
+\apiargument{IN}{nelems}{The number of elements in \source{}.  For
+    \FUNC{shmem\_team\_broadcast32}, this is the number of
+    32-bit halfwords. nelems must be of type \CTYPE{size\_t} in \Cstd.}
+\apiargument{IN}{PE\_root}{Zero-based ordinal of the \ac{PE}, with respect to
+    the team, from which the data is copied. Must be greater than or equal to
+    0 and less than N-1, where N is the result of calling \FUNC{shmem\_team\_n\_pes(team)}.
+    \VAR{PE\_root} must be of type \CTYPE{int}.} 
+
+\end{apiarguments}
+
+\apidescription{   
+    \openshmem broadcast routines are collective routines over an existing team.
+    They copy data object \source{} on the processor specified by \VAR{PE\_root}
+    and store the values at \dest{} on the other \acp{PE} that are members of the
+    team. The data is not copied to the \dest{} area on the root \ac{PE}.
+    
+    If the team has been created with the \LibConstRef{SHMEM\_TEAM\_NOCOLLECTIVE} option,
+    it will not have the required support structures to complete this routine. If
+    such a team is passed to this or any other team collective routine, the behavior
+    is undefined.
+
+    As with all \openshmem routines where the operation occurs for a given team -
+    either when the team is an argument to the routine, or when the team is an attribute
+    of the context argument to a routine - the \ac{PE} numbers are relative to the team,
+    and must be in the range of 0 to N-1, where N is the result of \FUNC{shmem\_team\_n\_pes(team)}.
+    
+    The values of the argument \VAR{PE\_root} must be the same value on all \acp{PE} in
+    the team. The same \dest{} and \source{} data objects must be passed by all \acp{PE}
+    in the team.
+
+    Upon return from a broadcast routine, the following are true for the local
+    \ac{PE}:
+    \begin{itemize}
+    \item If the current \ac{PE} is not the root \ac{PE},
+      the \dest{} data object is updated.
+    \item The \source{} data object may be safely reused.
+    \end{itemize}
+
+    Error checking will be done to ensure a valid team handle is provided.
+    All errors are considered fatal and will result in the job aborting with
+    an informative error message.
+}
+
+\apidesctable{
+The  \dest{}  and \source{} data  objects must conform to certain typing
+constraints, which are as follows:
+}{Routine}{Data type of \VAR{dest} and \VAR{source}}
+
+\apitablerow{shmem\_team\_broadcast64}{Any noncharacter
+    type that has an element size of \CONST{64} bits.
+    \CorCpp{} structures are NOT allowed.}
+\apitablerow{shmem\_team\_broadcast32}{Any noncharacter
+    type that has an element size of \CONST{32} bits.
+    \CorCpp{} structures are NOT allowed.}
+
+\apireturnvalues{
+    None.
+}
+
+\apinotes{
+    All \openshmem team collective routines use symmetric data structures associated
```

Comment:
Calling out the following?-
`The user must ensure that the symmetrically shared structures are not being updated by any PE in the team while any of the other PEs in the team participate in processing of an OpenSHMEM team broadcast routine.`

Reply:
Maybe this information belongs at the top of the section on collectives in a paragraph about team collectives?

Reply:
Makes sense, could also be applied to the non-team collectives as they are in v1.4 now.